src/test/java/com/example/produktapi/SeleniumTestCases.java

```diff
@@ -153,6 +154,141 @@ void addElectronicsProductAndCheckThatItIsAddedToTheCheckOut(){
+        ProductLink.click();
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
+
+        //3. Count number items
+        List<WebElement> productCategories = driver.findElements(By.className("col"));
+        int numberOfProducts = productCategories.size();
+        Assertions.assertEquals(4, numberOfProducts, "The number of items is not correct");
+    }
+
+    @Test
+    void MensProductAddToCart() {
+        //1. Click on top menu link "Shop"
+        driver.findElement(By.xpath("/html/body/header/div/div/ul/li[2]/a")).click();
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+        // 2. Click on Men's Clothing
+        WebElement ProductLink = driver.findElement(By.xpath("/html/body/div[1]/div/ul/li[2]/a"));
+        ProductLink.click();
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+
+
+        // 3. Find the Checkout button
+        var GetNumberOfProductsInCart = driver.findElement(By.xpath("//*[@id=\"buttonSize\"]")).getText();
+
+        //4. Check if cart was empty or not, if empty set value to 0.
+        int NumberInCart;
+        if(Objects.equals(GetNumberOfProductsInCart, "")) {
+            NumberInCart = 0;
+        }
+        else {
+            // 5. If cart is not empty then parse string value to integer.
+            NumberInCart = Integer.parseInt(GetNumberOfProductsInCart);
+        }
+        //6. increasing the product number by 1
+        int newNumberInCart = NumberInCart + 1;
+
+        // 7. Add a Men's Clothing product to cart
+        WebElement ProductAddToCart = driver.findElement(By.xpath("//*[@id=\"main\"]/div[1]/div/div/button"));
+        ProductAddToCart.click();
+
+        // 8. Get number of product in cart after adding a new product
+        var gettingNumberOfTheProductAfterAddingANew = driver.findElement(By.xpath("//*[@id=\"buttonSize\"]")).getText();
+        //9. Parsing the value from string to Int.
+        int numberOfTheProductInTheCartAfterAddingANew = Integer.parseInt(gettingNumberOfTheProductAfterAddingANew);
+        // 10. Verify that the product is added to the checkout.
+        Assertions.assertEquals(newNumberInCart,numberOfTheProductInTheCartAfterAddingANew );
+    }
+    @Test
+    void VerifyCheckOutButton() {
+        //1. Click on Checkout button
+        WebElement Checkout = driver.findElement(By.xpath("/html/body/header/div/div/div/a"));
+        Checkout.click();
+        // 2. Verify the heading of the Checkout Page
+        String CheckOutPageHeading = driver.findElement(By.tagName("h2")).getText();
+        Assertions.assertEquals("Checkout form", CheckOutPageHeading, "The heading is not correct");
+        System.out.println("Title of page is: " + CheckOutPageHeading);
+    }
+
+    @Test
+    void VerifyErrorMessageEmptyCheckOutForm()throws InterruptedException {
```

Comment:
Remove "throw InterrupterdException"

Reply:
Removed but added :"   try {
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }" 
The test case will not work without thread sleep. I tried "driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));" but I get error message "Element <button class="w-100 btn btn-primary btn-lg" type="submit"> could not be scrolled into view".